hydromt/data_catalog.py

```diff
@@ -32,60 +36,76 @@
 
-    def __init__(self, data_libs=None, logger=logger, **artifact_keys):
+    def __init__(
+        self, data_libs: Union[List, str] = [], logger=logger, **artifact_keys
+    ) -> None:
         """Catalog of DataAdapter sources to easily read from different files
         and keep track of files which have been accessed.
 
         Arguments
         ---------
         data_libs: (list of) str, Path, optional
-            One or more paths to yml files containing data sources which are parsed
-            to entries of the data catalog. By default the data catalog is initiated
-            without data entries. See :py:func:`~hydromt.data_catalog.DataCatalog.from_yml`
-            for accepted yml format.
+            One or more paths to data catalog yaml files or names of predefined data catalogs.
+            By default the data catalog is initiated without data entries.
+            See :py:func:`~hydromt.data_adapter.DataCatalog.from_yml` for accepted yaml format.
         artifact_keys:
-            key-word arguments specifying the name and version of a hydroMT data artifact,
-            to get the latest version use `True` instead of a version. For instance,
-            to get the latest data catalog with Deltares Data use `deltares_data=True`;
-            to get the latest
-
+            Deprecated from version v0.5
         """
+        if data_libs is None:  # legacy code. to be removed
+            data_libs = []
+        elif not isinstance(data_libs, list):  # make sure data_libs is a list
+            data_libs = np.atleast_1d(data_libs).tolist()
         self._sources = {}  # dictionary of DataAdapter
+        self._catalogs = {}  # dictionary of predefined Catalogs
         self._used_data = []
         self.logger = logger
-        for name, version in artifact_keys.items():
-            if version is None or not version:
+
+        # legacy code. to be removed
+        for lib, version in artifact_keys.items():
+            warnings.warn(
+                f"{lib}={version} as key-word argument is deprecated, add the predefined data catalog as string to the data_libs argument instead",
+                DeprecationWarning,
+            )
+            if not version:  # False or None
                 continue
-            if isinstance(version, str) and LooseVersion(version) <= LooseVersion(
-                "v0.0.4"
-            ):
-                raise ValueError("The minimal support version is v0.0.5")
-            self.from_artifacts(name=name, version=version)
-        if data_libs is not None:
-            for path in np.atleast_1d(data_libs):
-                self.from_yml(path)
+            elif isinstance(version, str):
+                lib += f"={version}"
+            data_libs = [lib] + data_libs
+
+        # parse data catalogs; both user and pre-defined
+        for name_or_path in data_libs:
+            if name_or_path.endswith(".yml"):  # user defined path
```

Comment:
Do we only support data_catalogs with .yml extension or should we also read in if .yaml ? Else we need to be very careful in the docs / error message.

Reply:
Good one, will make sure both extensions are recognized!